### PR TITLE
Final A/B-Tests

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Adds A/B compatible versions of the vsock and tcp throughput tests. Also includes the following small refractorings/improvements:
- Declare specific tests are particularly unstable. These are the tests that already have significant delta values associated with them in the current testing setup.
- Correctly associate p-values with metrics when submitting to Cloudwatch
- Correctly propagate CW metadata when reemitting captured logs (namespace and log stream name)
- Only run A/B tests if rust/python code was modified.
- Merge the two network tests into one file, because they are significantly simpler now

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/performance/test_vsock_ab.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        )
+        # The rootfs does not have iperf3-vsock
+        iperf3_guest = "/tmp/iperf3-vsock"
+
+        self._microvm.ssh.scp_put(self._iperf, iperf3_guest)
+        self._guest_iperf = iperf3_guest
+        return super().spawn_iperf3_client(client_idx)
+
+    def guest_command(self, port_offset):
+        return super().guest_command(port_offset).with_arg("--vsock")
```
Comment (zulinx86) [inline tests/integration_tests/performance/test_vsock_ab.py:68]: This looks completely same with the one in test_vsock_throughput.py. Can't we import it?

File: tests/integration_tests/performance/test_vsock_ab.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+LOAD_FACTOR = 1
+
+# Time (in seconds) for which iperf "warms up"
+WARMUP_SEC = 3
+
+# Time (in seconds) for which iperf runs after warmup is done
+RUNTIME_SEC = 20
+
+# VM guest memory size
+GUEST_MEM_MIB = 1024
```
Comment (zulinx86) [inline tests/integration_tests/performance/test_vsock_ab.py:25]: Can we move these constants inside `VsockIperf3Test` like what you do for `TCPIPerf3Test`?

**Review by zulinx86 — DISMISSED** [review]

File: tests/integration_tests/performance/test_vsock_ab.py (hunk 1 above)
Comment (roypat) [reply tests/integration_tests/performance/test_vsock_ab.py:68]: Mh, yeah, this is copied 1:1 from test_vsock_throughput.py. My philosophy with these was that I want the two test implementations to be completely independent, which is why I copied everything over (as test_vsock_throughput.py will hopefully go away soon).

File: tests/integration_tests/performance/test_vsock_ab.py (hunk 2 above)
Comment (roypat) [reply tests/integration_tests/performance/test_vsock_ab.py:25]: Sure! :)

**Review by pb8o — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Pull request merged**